### ISSUE TITLE
Mail seem to be corrupt after syncing to gmail account

### ISSUE THREAD

**Issue by rcbebo82** [imapsync/imapsync#391]
Good morning everybody.
I try to sync a imap mailbox from an ubuntu server with courier imap to gmail. The sync itself works without problem i use this command:
./imapsync --host1 87.230.26.* --user1 user1 --password1 password1 --host2 imap.gmail.com --user2 user2 --password2 password2 --gmail2 --nofoldersizes --skipsize

After the sync is complete all email appear without recipient and without subject in gmail webmail and in any email client. Any ideas therefore?
If i copy the mails manually using thunderbird or any other email client it works as expected. Currently google webmail appears as in the screenshot.
![image](https://user-images.githubusercontent.com/16867308/233567464-17be38a0-754c-4033-9614-b7e1c3b5971b.png)

**Comment by gilleslamiral** [comment]
This is strange.
Imapsync copies emails as they are, without changing a byte in them.
Use the --debugcontent option to see the content of the messages seen and transferred by imapsync.

Since we see "Received: " header lines, usually the first lines of a whole message, I suspect these are malformed emails starting with a non-header line like in the old Mbox format, a "From foo ..." line without the following colon.

Select one of these messages and look at it as a whole. Use a view-source in Thunderbird or a "Show original" in Gmail.

What do you see at the beginning?

**Comment by rcbebo82** [comment]
debugcontent output:
F message content begin next line (2120 characters long)
Delivery-date: Mon, 11 Mar 2019 01:38:14 +0100
Received: from [80.67.18.6] (helo=mx06.ispgateway.de)
        by mailcluster1-1.ispgateway.de with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
        (Exim 4.90_1)
        (envelope-from <service@musikschule-vierklang.de>)
        id 1h38xJ-0007DB-Sx; Mon, 11 Mar 2019 01:38:13 +0100
Return-path: <service@musikschule-vierklang.de>
X-Envelope-to: Service@musikschule-vierklang.de
Received: from [185.3.233.43] (helo=d10347.dedicated.alfahosting-pro.de)
        by mx06.ispgateway.de with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
        (Exim 4.90_1)
        (envelope-from <service@musikschule-vierklang.de>)
        id 1h38xJ-00070J-Ko
        for Service@musikschule-vierklang.de; Mon, 11 Mar 2019 01:38:13 +0100
Received: by d10347.dedicated.alfahosting-pro.de (Postfix, from userid 10003)
        id 49CC7109A08D; Mon, 11 Mar 2019 01:38:13 +0100 (CET)
Date: Mon, 11 Mar 2019 00:38:13 +0000
To: Service@musikschule-vierklang.de
From: Musikschule Vierklang <service@musikschule-vierklang.de>
Reply-To: sepi <sepi@saunalahti.fi>
Subject: Musikschule Vierklang: Application moderation # 0629027
Message-ID: <c1c62645830a08772578215b643e9dab@musikschule-vierklang.de>
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
X-Received-SPF: none ( mx06.ispgateway.de: domain of musikschule-vierklang.de does not provide an SPF record )
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
        spamfilter13.ispgateway.de
X-Spam-Level:
X-Spam-Status: No, hits=-6.6 required=9999.0 tests=BAYES_00,TVD_SPACE_RATIO autolearn=disabled
        version=3.4.0
X-Spam-CMAETAG: v=2.2 cv=NJulwwyg c=1 sm=1 tr=0
        a=TWXlzQN3dX2oQI5fifl+zQ==:17 a=IkcTkHD0fZMA:10 a=NTGMnVQrEZIA:10
        a=XVTA_rhIAAAA:8 a=h101GHb_AAAA:20 a=B2-WbguOf2O4EVzklAAA:9
        a=QEXdDO2ut3YA:10 a=Mr9uYPvWlalHWV62jlXS:22
X-Spam-CMAECATEGORY:
X-Spam-CMAESUBCATEGORY:
X-Spam-CMAESCORE:

Dies ist eine Mailanfrage via https://musikschule-vierklang.de/ von:
sepi <sepi@saunalahti.fi>

You have a new message. More details - https://24376.drive.google.com/open?id=1GjjUQSj7WTfq6TE3S38LeZWyjYPF8YSg

**Comment by gilleslamiral** [comment]
Well, if the header is precisely this one then it is broken. Header lines spanning several lines must continue with blanks at the beginning of the second line and the next ones. 

Is this message ok on the source side in IMAP?

**Comment by rcbebo82** [comment]
Original mail source:
Delivery-date: Mon, 11 Mar 2019 01:38:14 +0100

Received: from [80.67.18.6] (helo=mx06.ispgateway.de)

	by mailcluster1-1.ispgateway.de with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)

	(Exim 4.90_1)

	(envelope-from <service@musikschule-vierklang.de>)

	id 1h38xJ-0007DB-Sx; Mon, 11 Mar 2019 01:38:13 +0100

Return-path: <service@musikschule-vierklang.de>

X-Envelope-to: Service@musikschule-vierklang.de

Received: from [185.3.233.43] (helo=d10347.dedicated.alfahosting-pro.de)

	by mx06.ispgateway.de with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)

	(Exim 4.90_1)

	(envelope-from <service@musikschule-vierklang.de>)

	id 1h38xJ-00070J-Ko

	for Service@musikschule-vierklang.de; Mon, 11 Mar 2019 01:38:13 +0100

Received: by d10347.dedicated.alfahosting-pro.de (Postfix, from userid 10003)

	id 49CC7109A08D; Mon, 11 Mar 2019 01:38:13 +0100 (CET)

Date: Mon, 11 Mar 2019 00:38:13 +0000

To: Service@musikschule-vierklang.de

From: Musikschule Vierklang <service@musikschule-vierklang.de>

Reply-To: sepi <sepi@saunalahti.fi>

Subject: Musikschule Vierklang: Application moderation # 0629027

Message-ID: <c1c62645830a08772578215b643e9dab@musikschule-vierklang.de>

MIME-Version: 1.0

Content-Type: text/plain; charset=utf-8

X-Received-SPF: none ( mx06.ispgateway.de: domain of musikschule-vierklang.de does not provide an SPF record )

X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on

	spamfilter13.ispgateway.de

X-Spam-Level: 

X-Spam-Status: No, hits=-6.6 required=9999.0 tests=BAYES_00,TVD_SPACE_RATIO autolearn=disabled

	version=3.4.0

X-Spam-CMAETAG: v=2.2 cv=NJulwwyg c=1 sm=1 tr=0

	a=TWXlzQN3dX2oQI5fifl+zQ==:17 a=IkcTkHD0fZMA:10 a=NTGMnVQrEZIA:10

	a=XVTA_rhIAAAA:8 a=h101GHb_AAAA:20 a=B2-WbguOf2O4EVzklAAA:9

	a=QEXdDO2ut3YA:10 a=Mr9uYPvWlalHWV62jlXS:22

X-Spam-CMAECATEGORY: 

X-Spam-CMAESUBCATEGORY: 

X-Spam-CMAESCORE: 



Dies ist eine Mailanfrage via https://musikschule-vierklang.de/ von:

sepi <sepi@saunalahti.fi>



You have a new message. More details - https://24376.drive.google.com/open?id=1GjjUQSj7WTfq6TE3S38LeZWyjYPF8YSg

**Comment by rcbebo82** [comment]
Mail ist correctly displayed in Horde Webmail and Thinderbird. From, subject field are intact.

**Comment by gilleslamiral** [comment]
Well, your message source post is interpreted in github display. Put it like a source code, between four backquotes
 ````
like this
hello world
````

Anyway, if other clients display it normally, I'm getting out of ideas.
It may be related to newlines and how they are encoded `\r\n` vs `\r` vs `\n`
No clue.

**Comment by rcbebo82** [comment]
Updated the mail source. Hmm thats really strange maybe i can change some settings of courier imap service to get it working?

**Comment by rcbebo82** [comment]
Well i changed courier for dovecot and now mails seem to be correctly synced. Very strange.

**Comment by gilleslamiral** [comment]
> Maybe i can change some settings of courier imap service to get it working?

I'm not a courier expert.
A --debugcontent of the same message with both imap servers should give us a clue, examining both outputs closely.
But I do need a verbatim output, not a copy/paste. Two extracts from both logs, as files, would be great.

**Comment by rcbebo82** [comment]
Well i want to know what is causing the problem and you also i guess. If the migration with dovecot is completed i would switch back to courier and provide the debug outputs then. Maybe this would help anyone in the future!

**Comment by gilleslamiral** [comment]
> Well i want to know what is causing the problem and you also i guess

No, I don't care. If you found a solution that's ok to me, I'm just here to help you

**Comment by acardielf** [comment]
I'm having same issue. I think it's related to `by xxxx.xxx with esmtp (Exim 4.60)` in header. 

If someone has solution, please, let me know :) Thanks!

**Comment by gilleslamiral** [comment]
> I'm having same issue. I think it's related to `by xxxx.xxx with esmtp (Exim 4.60)` in header.
> 
> If someone has solution, please, let me know :) Thanks!

What happens if you transfer a message with imapsync to an other IMAP account than Gmail?
